src/components/AuthContainer.js

```
@@ -54,33 +84,70 @@ export default class AuthContainer extends Component {
+      div({ style: { marginTop: '3rem', display: 'flex' } }, [
+        div({ style: { lineHeight: '170%' } }, ['First Name', br(),
+          textInput({
+            value: this.state.givenName,
+            onChange: e => this.setState({ givenName: e.target.value })
+          })]),
+        div({ style: { width: '1rem' } }),
+        div({ style: { lineHeight: '170%' } }, ['Last Name', br(), textInput({
+          value: this.state.familyName,
+          onChange: e => this.setState({ familyName: e.target.value })
+        })])
+      ]),
+      div({ style: { lineHeight: '170%' } }, [
+        div({ style: { marginTop: '2rem' } }, 'Contact Email for Notifications'),
+        div({}, [textInput({
+          value: this.state.email,
+          onChange: e => this.setState({ email: e.target.value }),
+          style: { width: '50ex' }
+        })])
+      ]),
+      div({ style: { marginTop: '3rem' } }, [
+        buttonPrimary({ disabled: this.state.busy, onClick: () => this.register() },
+          'Register'
+        ),
+        this.state.busy && spinner({
+          size: 34, style: { display: null, margin: null, marginLeft: '1ex' }
+        })
+      ])
+    ])
+  }
+
+  async register() {
+    this.setState({ busy: true })
+    await Sam.createUser()
+    const blankProfile = {
+      firstName: 'N/A',
+      lastName: 'N/A',
+      title: 'N/A',
+      contactEmail: 'N/A',
+      institute: 'N/A',
+      institutionalProgram: 'N/A',
+      programLocationCity: 'N/A',
+      programLocationState: 'N/A',
+      programLocationCountry: 'N/A',
+      pi: 'N/A',
+      nonProfitStatus: 'N/A'
+    }
+    const firstName = this.state.givenName
+    const lastName = this.state.familyName
+    const contactEmail = this.state.email
+    await Orchestration.profile().set(_.merge(blankProfile, { firstName, lastName, contactEmail }))
+    this.setState({ busy: false, isRegistered: true })
```

What happens if registration fails?

---

Silent error. I'm hoping that #127 is forthcoming. Should we still handle these in the meantime?

---

That's fine for now, and the global error handling should cover it.